File: bank.cpp
#include "config.h"
#include "bank.h"

BankAccount::BankAccount(int aNumber) {
  accountNumber = aNumber;
  funds = 0;
}

// Current ID is for the next account in line. If accounts get
// removed, this will ensure people can keep their account number.
// but never more than maxAccounts at a time
Bank::Bank() {
  numOfAccounts = 0;
  currentID = 0;
  bankAccount = new BankAccount*[config::maxAccounts];
}

Bank::~Bank() {
  for(int i=0; i < numOfAccounts; i++)
    delete bankAccount[i];
  delete []bankAccount;
}

int Bank::createAccount() {
  if(numOfAccounts >= config::maxAccounts)
    return -1;
  bankAccount[numOfAccounts] = new BankAccount(currentID);
  currentID++;
  return numOfAccounts++;
}

// TODO:: This needs a hashmap or similar eventually so that when
// account numbers =/= the index it'll still find them in O(1) time.
int Bank::displayFunds(int aNumber) {
  return bankAccount[aNumber]->funds;
}

//TODO: Eventually design this into a credit/debit transaction
// to ensure all money is kept stable potentially?
void Bank::addFunds(int aNumber, int funds) {
  bankAccount[aNumber]->funds = funds;
}

File: bank.py
class BankAccount:
    def __init__(self, accountNumber):
        self.funds = 0
        self.acountNumber = accountNumber

class Bank:
    def __init__(self, bankID):
        self.id = bankID
        self.currentID = 0 # This is the current available account ID
        self.accounts = {}

    def createAccount(self):
        self.accounts[self.currentID] = BankAccount(self.currentID)
        self.currentID += 1
        return [self.id, self.currentID - 1] # Currently saves as [bank, account]

    def displayFunds(self, account):
        return self.accounts[account].funds

    def addFunds(self, account, funds):
        self.accounts[account].funds += funds
